pkgs/sdk/server-ai/src/LdAiClient.cs

```diff
@@ -0,0 +1,172 @@
+            {
+                if (kvp.Key == LdContextVariable)
+                {
+                    _client.GetLogger().Warn("AI model config variables contains 'ldctx' key, which is reserved; this key will be the value of the LaunchDarkly context");
+                    continue;
+                }
+                mergedVariables[kvp.Key] = kvp.Value;
+            }
+        }
+
+
+        var prompt =
+            parsed.Prompt?.Select(m => new LdAiConfig.Message(InterpolateTemplate(m.Content, mergedVariables), m.Role));
+
+        return new LdAiConfigTracker(_client, key, new LdAiConfig(parsed.Meta?.Enabled ?? false, prompt, parsed.Meta, parsed.Model), context);
+    }
+
+    /// <summary>
+    /// Retrieves all attributes from the given context, including private attributes. The attributes
+    /// are converted into C# primitives recursively.
+    /// </summary>
+    /// <param name="context">the context</param>
+    /// <returns>the attributes</returns>
+    private static IDictionary<string, object> GetAllAttributes(Context context)
+    {
+        var attributes = new Dictionary<string, object>();
+        foreach (var key in context.OptionalAttributeNames)
+        {
+            attributes[key] = ValueToObject(context.GetValue(AttributeRef.FromLiteral(key)));
+        }
+
+        attributes["kind"] = context.Kind.ToString();
+        attributes["key"] = context.Key;
+        attributes["anonymous"] = context.Anonymous;
+
+        return attributes;
+    }
+
+    /// <summary>
+    /// Recursively converts an LdValue into a C# object.
+    /// </summary>
+    /// <param name="value">the LdValue</param>
+    /// <returns>the object</returns>
+    private static object ValueToObject(LdValue value)
+    {
+        return value.Type switch
+        {
+            LdValueType.Null => null,
+            LdValueType.Bool => value.AsBool,
+            LdValueType.Number => value.AsDouble,
+            LdValueType.String => value.AsString,
+            LdValueType.Array => value.List.Select(ValueToObject).ToList(),
+            LdValueType.Object => value.Dictionary
+                .Select(kv => new KeyValuePair<string, object>(kv.Key, ValueToObject(kv.Value)))
+                .ToImmutableDictionary(),
+            _ => null
+        };
+    }
+
+
+    private static string InterpolateTemplate(string template, IReadOnlyDictionary<string, object> variables)
+    {
+        return Template.Compile(template).Render(variables);
+    }
+
+
+    private AiConfig ParseConfig(LdValue value, string key)
+    {
+
+        var serialized = value.ToJsonString();
+        try
+        {
+            return JsonSerializer.Deserialize<AiConfig>(serialized);
+        }
+        catch (JsonException e)
+        {
+            _client.GetLogger().Error("Unable to parse AI model config for key {0}: {1}", key, e.Message);
+            return null;
+        }
+    }
+
+    /// <summary>
+    /// THD
+    /// </summary>
+    public void Dispose()
```

Comment:
Unclear if this should really be implemented or not. If you are instantiating the client directly with a server client, then this makes sense.

If you are just wrapping it, and might use the base client for other stuff, then no.